crypto/crypto.go

```diff
@@ -205,6 +205,10 @@ func Sign(hash []byte, prv *ecdsa.PrivateKey) (sig []byte, err error) {
 }
 
 func Encrypt(pub *ecdsa.PublicKey, message []byte) ([]byte, error) {
+	if pub.X == nil || pub.Y == nil {
+		return nil, errors.New("invalid public key")
+	}
```

Comment:
Hmm, could you detail where this scenario occurred that the public key was invalid?